src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets

```diff
@@ -58,6 +58,14 @@
 
     <XunitOptions>$(XunitOptions) -notrait category=non$(_bc_TargetGroup)tests</XunitOptions>
 
+    <TargetOS Condition="'$(TargetOS)' == ''">$(DefaultOSGroup)</TargetOS>
+    <TargetOSTrait Condition="'$(TargetOS)'=='Windows_NT'">nonwindowstests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(TargetOS)'=='Linux'">nonlinuxtests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(TargetOS)'=='OSX'">nonosxtests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(TargetOS)'=='FreeBSD'">nonfreebsdtests</TargetOSTrait>
+    <TargetOSTrait Condition="'$(TargetOS)'=='NetBSD'">nonnetbsdtests</TargetOSTrait>
+    <XunitOptions Condition="'$(TargetOSTrait)' != ''">$(XunitOptions) -notrait category=$(TargetOSTrait)</XunitOptions>
```

Comment:
Should we instead of appending it to `XunitOptions` add it to `XunitTraitOptions` that is added to TestCommandLine through the placeholder replacement here: https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L191 ?